moped-editor/src/views/projects/newProjectView/NewProjectMap.js

```diff
@@ -454,9 +454,11 @@ const NewProjectMap = ({
         maxZoom={20}
         width="100%"
         height="60vh"
-        interactiveLayerIds={!isDrawing && getEditMapInteractiveIds()}
-        onHover={!isDrawing ? handleLayerHover : null}
-        onClick={!isDrawing ? handleLayerClick : null}
+        interactiveLayerIds={
+          !isDrawing && !isSignalComponent && getEditMapInteractiveIds()
+        }
+        onHover={!isDrawing && !isSignalComponent ? handleLayerHover : null}
+        onClick={!isDrawing && !isSignalComponent ? handleLayerClick : null}
```

Comment:
these click and hover events must be disabled for signal components because we're not rendering those interactive layers. we could maybe avoid some of this condition checking by refactoring the click/hover handlers.

Reply:
I was wondering what would happen if a component changes type from signal to something else, but I think the code you have written should handle it just fine.